pkg/operator/operator.go

```diff
@@ -0,0 +1,261 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"context"
+	"crypto/x509"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	nodeobservabilityv1alpha1 "github.com/openshift/node-observability-operator/api/v1alpha1"
+	nodeobservabilityv1alpha2 "github.com/openshift/node-observability-operator/api/v1alpha2"
+	operatorconfig "github.com/openshift/node-observability-operator/pkg/operator/config"
+	machineconfigcontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/machineconfig"
+	nodeobservabilitycontroller "github.com/openshift/node-observability-operator/pkg/operator/controller/nodeobservability"
+	nodeobservabilityrun "github.com/openshift/node-observability-operator/pkg/operator/controller/nodeobservabilityrun"
+)
+
+const (
+	operatorName = "node_observability_operator"
+)
+
+var (
+	setupLog = ctrl.Log.WithName("operator")
+)
+
+// Client holds the API clients required by Operator.
+type Client struct {
+	client.Client
+	meta.RESTMapper
+}
+
+// Operator hold the client, manager, and logging resources
+// for the nodeobservability opreator.
+type Operator struct {
+	client  Client
+	manager manager.Manager
+	log     logr.Logger
+}
+
+// New creates a new operator from cliCfg and opCfg.
+func New(cliCfg *rest.Config, opCfg *operatorconfig.Config) (*Operator, error) {
+	opts := zap.Options{
+		TimeEncoder: zapcore.TimeEncoder(func(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
+			enc.AppendString(t.UTC().Format("2006-01-02T15:04:05.000Z"))
+		}),
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+	logger := zap.New(zap.UseFlagOptions(&opts))
+	ctrl.SetLogger(logger)
```

Comment:
yes, but then I wouldn't be able to access the logger variable for the manager in operator.go

Reply:
GitHub didn't show all the lines I selected, I mean all this can be moved to `main.go`:
```
	opts := zap.Options{
		TimeEncoder: zapcore.TimeEncoder(func(t time.Time, enc zapcore.PrimitiveArrayEncoder) {
			enc.AppendString(t.UTC().Format("2006-01-02T15:04:05.000Z"))
		}),
	}
	opts.BindFlags(flag.CommandLine)
	flag.Parse()
	logger := zap.New(zap.UseFlagOptions(&opts))
	ctrl.SetLogger(logger)
```

The logger set by controller-runtime's global function `SetLogger` is global too, so you can use it from any place in the code.